.github/workflows/update-subtree.yml

```diff
@@ -0,0 +1,155 @@
+        CURRENT_COMMIT_HASH=$(curl https://static.rust-lang.org/dist/$CURRENT_TOOLCHAIN_DATE/channel-rust-nightly-git-commit-hash.txt)
+        NEXT_COMMIT_HASH=$(curl https://static.rust-lang.org/dist/$NEXT_TOOLCHAIN_DATE/channel-rust-nightly-git-commit-hash.txt)
+        if [ -z "$CURRENT_COMMIT_HASH" ]; then
+          echo "Could not find current commit hash on static.rust-lang.org"
+          exit 1
+        fi
+        if [ -z "$NEXT_COMMIT_HASH" ]; then
+          echo "Could not find next commit hash on static.rust-lang.org"
+          exit 1
+        fi
+        echo "CURRENT_TOOLCHAIN_DATE=${CURRENT_TOOLCHAIN_DATE}" >> $GITHUB_ENV
+        echo "NEXT_TOOLCHAIN_DATE=${NEXT_TOOLCHAIN_DATE}" >> $GITHUB_ENV
+        echo "CURRENT_COMMIT_HASH=${CURRENT_COMMIT_HASH}" >> $GITHUB_ENV
+        echo "NEXT_COMMIT_HASH=${NEXT_COMMIT_HASH}" >> $GITHUB_ENV
+
+    - name: Update subtree/library locally
+      run: |
+        cd rust-tmp
+
+        # Ensure "upstream/master" branch contains the target commit
+        if ! git show ${NEXT_COMMIT_HASH} --oneline --no-patch; then
+          echo "Rust commit ${NEXT_COMMIT_HASH} cannot be found."
+          exit 1
+        fi
+
+        git checkout ${NEXT_COMMIT_HASH}
+        ../git-filter-repo/git-filter-repo --subdirectory-filter library --force
+        git checkout -b subtree/library
+
+        cd ../verify-rust-std
+        git remote add rust-filtered ../rust-tmp/
+        git fetch rust-filtered
+        git checkout -b subtree/library rust-filtered/subtree/library
+        SUBTREE_HEAD_MSG=$(git log --format=%s -n 1 origin/subtree/library)
+        UPSTREAM_FROM=$(git log --grep="${SUBTREE_HEAD_MSG}" -n 1 --format=%H rust-filtered/subtree/library)
+        UPSTREAM_HEAD=$(git log --format=%H -n 1 rust-filtered/subtree/library)
+        if [ "${UPSTREAM_HEAD}" = "${UPSTREAM_FROM}" ]; then
+          echo "Nothing to do, ${UPSTREAM_FROM} matches ${UPSTREAM_HEAD} (${SUBTREE_HEAD_MSG})"
+          echo "MERGE_CONFLICTS=noop" >> $GITHUB_ENV
+        else
+          git branch --set-upstream-to=origin/subtree/library
+          git -c user.name=gitbot -c user.email=git@bot rebase
+          echo "MERGE_CONFLICTS=maybe" >> $GITHUB_ENV
+        fi
+
+    - name: Create Pull Request
+      if: ${{ env.MERGE_CONFLICTS != 'noop' }}
+      uses: peter-evans/create-pull-request@v7
+      with:
+        title: 'Update subtree/library to ${{ env.NEXT_TOOLCHAIN_DATE }}'
+        body: |
+          This is an automated PR to update the subtree/library branch to the changes
+          from ${{ env.CURRENT_TOOLCHAIN_DATE }} (${{ env.CURRENT_COMMIT_HASH }})
+          to ${{ env.NEXT_TOOLCHAIN_DATE }} (${{ env.NEXT_COMMIT_HASH }}), inclusive.
+        branch: update-subtree/library
+        delete-branch: true
+        base: subtree/library
+        path: verify-rust-std
+
+    - name: Merge subtree/library changes
+      if: ${{ env.MERGE_CONFLICTS != 'noop' }}
```

Comment:
Could we exit early if the update is a no-op to avoid create a PR in the next job?

Reply:
I believe we are implicitly doing this: the next steps require that `MERGE_CONFLICTS` is either `yes` or `no`, so leaving the value at `noop` will mean they are not considered.